sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -5243,11 +5243,11 @@ class AstBuilder extends DataTypeAstBuilder
       } else {
         None
       }
-      val setDefaultExpression: Option[String] =
+      val setDefaultExpression: Option[DefaultValueExpression] =
         if (action.defaultExpression != null) {
-          Option(action.defaultExpression()).map(visitDefaultExpression).map(_.originalSQL)
+          Option(action.defaultExpression()).map(visitDefaultExpression)
         } else if (action.dropDefault != null) {
-          Some("")
+          Some(DefaultValueExpression(Literal(null), ""))
```

Comment:
This seems a bit fragile. I wonder whether there is a better way to handle it. We can have an enum or something with unchanged, drop, and set options. It only makes sense, however, if we want to pass null for `DefaultValue` to indicate drop in the public connector API.

Reply:
Fixed in the latest pr, add a boolean dropDefault in AlterColumnSpec